src/mappings/Controller/mapping.ts

```diff
@@ -270,106 +280,142 @@ export function handleRemoveGlobalConstraint(
-                                       vmAddress: Address,
-                                       vmParamsHash: Bytes,
-                                       contractToCall: Bytes): void {
-   setGPParams(vmAddress, vmParamsHash);
-   let controllerScheme =  ControllerScheme.load(crypto.keccak256(concat(avatar, scheme)).toHex());
-   let genericSchemeParams = new GenericSchemeParam(scheme.toHex());
-   genericSchemeParams.votingMachine = vmAddress;
-   genericSchemeParams.voteParams = vmParamsHash.toHex();
-   genericSchemeParams.contractToCall = contractToCall;
-   genericSchemeParams.save();
-   controllerScheme.genericSchemeParams = genericSchemeParams.id;
-   controllerScheme.save();
+export function setGenericSchemeParams(
+  avatar: Address,
+  scheme: Address,
+  vmAddress: Address,
+  vmParamsHash: Bytes,
+  contractToCall: Bytes,
+): void {
+  setGPParams(vmAddress, vmParamsHash);
+  let controllerScheme = ControllerScheme.load(
+    crypto.keccak256(concat(avatar, scheme)).toHex(),
+  );
+  let genericSchemeParams = new GenericSchemeParam(scheme.toHex());
+  genericSchemeParams.votingMachine = vmAddress;
+  genericSchemeParams.voteParams = vmParamsHash.toHex();
+  genericSchemeParams.contractToCall = contractToCall;
+  genericSchemeParams.save();
+  controllerScheme.genericSchemeParams = genericSchemeParams.id;
+  if (controllerScheme != null) {
+    controllerScheme.save();
+  }
 }
 
-export function setUGenericSchemeParams(avatar: Address,
-                                        scheme: Address,
-                                        vmAddress: Address,
-                                        vmParamsHash: Bytes,
-                                        contractToCall: Bytes): void {
-   setGPParams(vmAddress, vmParamsHash);
-   let controllerScheme =  ControllerScheme.load(crypto.keccak256(concat(avatar, scheme)).toHex());
-   let genericSchemeParams = new UGenericSchemeParam(controllerScheme.paramsHash.toHex());
-   genericSchemeParams.votingMachine = vmAddress;
-   genericSchemeParams.voteParams = vmParamsHash.toHex();
-   genericSchemeParams.contractToCall = contractToCall;
-   genericSchemeParams.save();
-   controllerScheme.uGenericSchemeParams = genericSchemeParams.id;
-   controllerScheme.save();
+export function setUGenericSchemeParams(
+  avatar: Address,
+  scheme: Address,
+  vmAddress: Address,
+  vmParamsHash: Bytes,
+  contractToCall: Bytes,
+): void {
+  setGPParams(vmAddress, vmParamsHash);
+  let controllerScheme = ControllerScheme.load(
+    crypto.keccak256(concat(avatar, scheme)).toHex(),
+  );
+  let genericSchemeParams = new UGenericSchemeParam(
+    controllerScheme.paramsHash.toHex(),
+  );
+  genericSchemeParams.votingMachine = vmAddress;
+  genericSchemeParams.voteParams = vmParamsHash.toHex();
+  genericSchemeParams.contractToCall = contractToCall;
+  genericSchemeParams.save();
+  controllerScheme.uGenericSchemeParams = genericSchemeParams.id;
+  if (controllerScheme != null) {
```

Comment:
You're accessing `controllerScheme` here before checking whether it is `null`. I've seen the same in a few other places. If `controllerScheme` is null, line 417 may already fail.